### ISSUE TITLE
Message handler not fired for blank message in QOS mode

### ISSUE THREAD

**Issue by witq** [dropbox/amqp-coffee#72]
Hi, I have an issue where a blank message (0 size, empty body) published to a queue I'm consuming with amqp-coffee stops the consumer.
I'm using QOS mode to validate each message and then reject it or process and ack it. When a blank message gets published to the queue, my message handler is not fired and the message can be neither acked nor rejected – the consumer just stops working. When I disable prefetchCount, the empty messages are discarded silently.
My guess is that somewhere a callback is not called and everything just stops. I tried debugging it but couldn't get anywhere. Maybe it's something you can look into? I'll gladly help.

**Comment by barshow** [comment]
Can you add a failing test? or write up some pseudo code?  Thanks!

**Comment by witq** [comment]
Hi, thank you for responding. I did't have time to do it earlier but here's a failing test for you (add to consumer.test.coffee):

```
it 'test we can consume an empty message', (done)->
    amqp = null
    queue = uuid()

    messageProcessor = (m)->
      done()

    async.series [
      (next)->
        amqp = new AMQP { host:'localhost' }, (e, r)->
          should.not.exist e
          next()

      (next)->
        amqp.queue {queue}, (e,q)->
          q.declare ()->
            q.bind "amq.direct", queue, next

      (next)->
        amqp.publish "amq.direct", queue, undefined, {confirm: true}, next

      (next)->
        amqp.consume queue, { prefetchCount: 10 }, messageProcessor, (e,r)->
          should.not.exist e
          next()
    ]
```
As far as I can see, disabling prefetch count when setting up the consumer has no effect, the test always fails after the timeout expires. I hope this helps.

**Comment by barshow** [comment]
Thank you for writing this!  This should now be fixed in 0.1.30  Let me know if there are still issues.
